### PR TITLE
Pubmed configuration files

### PR DESCRIPTION
**Pull request by gnott**
These two new files - pubmed.cfg and publication_types.yaml - will be requirements for deploying the new PubMed code. I wasn't sure about the underscores in file names and how you'd like them in salt.

I don't know why this has 5 commits, since I based the branch (twice now) on the current master branch. The result is the same, I guess if you just review the diff.

### CODE REVIEW

File: salt/elife-bot/init.sls [hunk 1]
```diff
@@ -63,6 +63,22 @@ elife-bot-crossref-cfg:
         - require:
             - elife-bot-repo
 
+elife-bot-pubmed-cfg:
+    file.managed:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /opt/elife-bot/pubmed.cfg
+        - source: salt://elife-bot/config/opt-elife-bot-pubmed.cfg
```
Comment (giorgiosironi) [inline salt/elife-bot/init.sls]: this is the expected convention, it's simply the desired path with `/` replaced by `-`

File: salt/elife-bot/init.sls [hunk 2]
```diff
@@ -63,6 +63,22 @@ elife-bot-crossref-cfg:
         - require:
             - elife-bot-repo
 
+elife-bot-pubmed-cfg:
+    file.managed:
+        - user: {{ pillar.elife.deploy_user.username }}
+        - name: /opt/elife-bot/pubmed.cfg
```
Comment (giorgiosironi) [inline salt/elife-bot/init.sls]: make sure these files are ignored in the `.gitignore` of elife-bot

**Comment by giorgiosironi** [comment]
A fresh stack was created successfully, but the stack with the current `master` failed afterwards. Reproducing now, I think I'll remove bucket/topics from the `ci` stack if they caused this problem.

**Comment by giorgiosironi** [comment]
`elife-bot-formula-pr-16-elife-published already exists` is the error. We have seen this before.

**Comment by giorgiosironi** [comment]
```
[11:48:08][giorgio@Newton:~/code/builder]$ aws s3 ls | grep formula
2017-09-29 23:16:23 elife-bot-formula-pr-10-elife-published
2017-10-12 02:47:36 elife-bot-formula-pr-11-elife-published
2017-11-06 15:11:25 elife-bot-formula-pr-12-elife-published
2017-12-05 06:30:45 elife-bot-formula-pr-13-elife-published
2017-12-06 00:51:27 elife-bot-formula-pr-14-elife-published
2017-12-06 02:08:45 elife-bot-formula-pr-15-elife-published
2017-12-14 00:49:31 elife-bot-formula-pr-16-elife-published
2017-12-05 09:35:04 elife-bot-formula-pr-7-elife-published
2017-09-27 16:52:07 elife-bot-formula-pr-9-elife-published
```
This bucket probably has a retain policy.

**Comment by giorgiosironi** [comment]
https://github.com/elifesciences/builder/commit/54fce17ef053088523c5ef87fab1c38b7f20eedd means we should not see any bucket being created/removed in these tests. The build is about the Salt formula so unless it actually has to use the buckets during the highstate, it would just be a slowdown anyway.

**Comment by giorgiosironi** [comment]
I think builder is merging `s3` from the project's base environment and creating the buckets anyway.

**Comment by giorgiosironi** [comment]
I think builder is merging `s3` from the project's base environment and creating the buckets anyway. I'll see if I can start by overriding their deletion policies rather than removing them, for the moment.

File: salt/elife-bot/init.sls (hunk 1 above)
Comment (lsh-0) [reply salt/elife-bot/init.sls]: yes, it's just something I made up to keep the namespace shallow and it's contents obvious. it's not enforced by salt or anything

**Comment by gnott** [comment]
The correct commit order I guess would be if you can please review and merge https://github.com/elifesciences/elife-bot/pull/571 which will have git ignore these files, then the files can be added by this PR.

**Pull request merged**